pkg/kubelet/pod_manager.go

```diff
@@ -127,13 +127,13 @@ func (self *basicPodManager) UpdatePods(u PodUpdate, podSyncTypes map[types.UID]
 }
 
 // Set the internal pods based on the new pods.
-func (self *basicPodManager) SetPods(newPods []api.Pod) {
-	self.lock.Lock()
-	defer self.lock.Unlock()
-	self.setPods(newPods)
+func (pm *basicPodManager) SetPods(newPods []api.Pod) {
+	pm.lock.Lock()
```

Comment:
We need to be careful about using public embedded field. If we embed lock, lock would be a public field. People can call pm.Lock() outside the pkg.